File: server/src/main/java/com/ahmadramadhan/mudahtitip/entities/AgreementStatus.java
package com.ahmadramadhan.mudahtitip.entities;

/**
 * Status of an agreement negotiation.
 */
public enum AgreementStatus {
    /**
     * Initial proposal, waiting for response.
     */
    PROPOSED,

    /**
     * Counter-offer made, waiting for response.
     */
    COUNTER,

    /**
     * Agreement accepted by both parties.
     */
    ACCEPTED,

    /**
     * Agreement rejected, no deal.
     */
    REJECTED
}

File: client/lib/features/agreement/data/models/agreement_status.dart
/// Status of an agreement negotiation.
enum AgreementStatus {
  /// Initial proposal, waiting for response.
  proposed('PROPOSED'),

  /// Counter-offer made, waiting for response.
  counter('COUNTER'),

  /// Agreement accepted by both parties.
  accepted('ACCEPTED'),

  /// Agreement rejected, no deal.
  rejected('REJECTED');

  final String value;
  const AgreementStatus(this.value);

  /// Get display name in Indonesian.
  String get displayName => switch (this) {
    AgreementStatus.proposed => 'Menunggu Respons',
    AgreementStatus.counter => 'Penawaran Balik',
    AgreementStatus.accepted => 'Disetujui',
    AgreementStatus.rejected => 'Ditolak',
  };

  /// Parse from backend string value.
  static AgreementStatus fromString(String value) {
    return AgreementStatus.values.firstWhere(
      (s) => s.value == value.toUpperCase(),
      orElse: () => AgreementStatus.proposed,
    );
  }
}
